File: go_lottoresult/app/utils/lotto-utils.go
package utils

import "time"

type LottoUtil struct{}

var LottoUtilObj = LottoUtil{}

var _START_DATE = DatUtilObj.GetKoreanDateFromYmd(2002, 12, 7)

const _DRAW_HOUR_MIN = 2045

func isBeforeDrawTime(date time.Time) bool {
	return date.Hour()*100+date.Minute() < _DRAW_HOUR_MIN
}

func (l LottoUtil) GetLastDrawNo() int {
	curDate := DatUtilObj.GetKoreanDate()
	drwNo := DatUtilObj.WeekDiff(_START_DATE, curDate) + 1
	if curDate.Weekday() != time.Saturday || !isBeforeDrawTime(curDate) {
		return drwNo
	}
	return drwNo - 1
}

func (l LottoUtil) GetMaxDrawNo() int {
	curDate := DatUtilObj.GetKoreanDate()
	drwNo := DatUtilObj.WeekDiff(_START_DATE, curDate) + 1
	if curDate.Weekday() != time.Saturday {
		return drwNo
	}
	return drwNo + 1
}

func (l LottoUtil) GetDrawDate(drwNo int) time.Time {
	return DatUtilObj.WeekAdd(_START_DATE, drwNo-1)
}

File: flutter_lottoresult/lib/system/utils/lotto_utils.dart
import 'package:flutter_lottoresult/system/extensions/tz_datetime_extensions.dart';
import 'package:flutter_lottoresult/system/utils/date_utils.dart';
import 'package:timezone/timezone.dart';

class LottoUtils {
  static final _startDate = DateUtils.getKoreanTimeFromYmd(2002, 12, 7);
  static final _drawHourMin = 2045;

  static bool isBeforeDrawTime(TZDateTime time) {
    return time.hour * 100 + time.minute < _drawHourMin;
  }

  static int getLastDrawNo() {
    final curTime = DateUtils.getKoreanTime();
    final drwNo = DateUtils.weekDiff(_startDate, curTime) + 1;
    if (curTime.weekday != DateTime.saturday || isBeforeDrawTime(curTime)) {
      return drwNo;
    }
    return drwNo - 1;
  }

  static int getMaxDrawNo() {
    final curTime = DateUtils.getKoreanTime();
    final drwNo = DateUtils.weekDiff(_startDate, curTime);
    if (curTime.weekday != DateTime.saturday) {
      return drwNo;
    }
    return drwNo + 1;
  }

  static TZDateTime getDrawDate(int drwNo) {
    return _startDate.weekAdd(drwNo - 1);
  }
}
